### ISSUE TITLE
PhysicalX / PhysicalY are incorrect in web engine when clicking on top of flt-text-editing-host 

### ISSUE THREAD

**Issue by jezell** [flutter/flutter#125948]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

1. Add a widget to the screen that has text editing. Position it somewhere so that the left and top edges are in the middle of the screen.
2. Set a breakpoint in PointerEventConverter on case ui.PointerChange.down
3. Without the textbox selected, click inside the text control and note the physicalX
4. Click in the same spot and note that the physicalX is much different.

The engine seems to assume that the flutter view was the source of all events, so when it hits flt-text-editing-host it delivers incorrect pointer event data. 

### Proposal

Perhaps the web engine should make sure that the event was actually on the canvas host element it thinks it was on, and if not, convert the coordinates when converting the DomEvent to PointerData.

**Comment by jezell** [comment]
A temporary hack for anyone encountering this is that you can set display:none to flt-text-editing-host, but maybe that has undesirable effects in some cases (not popping keyboard on mobile browsers?).

**Comment by darshankawar** [comment]
Thanks for the report @jezell 

> ### Use case
> 1. Add a widget to the screen that has text editing. Position it somewhere so that the left and top edges are in the middle of the screen.
> 2. Set a breakpoint in PointerEventConverter on case ui.PointerChange.down
> 3. Without the textbox selected, click inside the text control and note the physicalX
> 4. Click in the same spot and note that the physicalX is much different.

Can you provide us a minimal complete code sample for this ?

**Comment by jezell** [comment]
@darshankawar yeah, I'll try to put one together. Was seeing it inside our app with unreliable selection when zoomed and took a bit to track down why. I think I should be able to put something together that illustrates the problem now that I know the cause.

**Comment by jezell** [comment]
This sample illustrates the problem. Click twice over the textbox, the first click will hit the canvas. The second click should hit the text editing host. Note that it prints widely different coordinates for the local positionX despite hitting the same position:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/gestures.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // TRY THIS: Try running your application with "flutter run". You'll see
        // the application has a blue toolbar. Then, without quitting the app,
        // try changing the seedColor in the colorScheme below to Colors.green
        // and then invoke "hot reload" (save your changes or press the "hot
        // reload" button in a Flutter-supported IDE, or press "r" if you used
        // the command line to start the app).
        //
        // Notice that the counter didn't reset back to zero; the application
        // state is not lost during the reload. To reset the state, use hot
        // restart instead.
        //
        // This works for code too, not just values: Most code changes can be
        // tested with just a hot reload.
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final scale = Matrix4.identity();
    scale.translate(-100, -100);
    scale.scale(2);

    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Material(
        child: Stack(
            // Center is a layout widget. It takes a single child and positions it
            // in the middle of the parent.
            // Column is also a layout widget. It takes a list of children and
            // arranges them vertically. By default, it sizes itself to fit its
            // children horizontally, and tries to be as tall as its parent.
            //
            // Column has various properties to control how it sizes itself and
            // how it positions its children. Here we use mainAxisAlignment to
            // center the children vertically; the main axis here is the vertical
            // axis because Columns are vertical (the cross axis would be
            // horizontal).
            //
            // TRY THIS: Invoke "debug painting" (choose the "Toggle Debug Paint"
            // action in the IDE, or press "p" in the console), to see the
            // wireframe for each widget.

            children: [
          Transform(
              alignment: Alignment.topLeft,
              transform: scale,
              child: Stack(children: [
                Positioned(
                    top: 100,
                    left: 100,
                    child: SizedBox(
                        width: 4000,
                        height: 4000,
                        child: ClipRect(
                            child: Listener(
                                onPointerDown: (event) =>
                                    print(event.localPosition),
                                child: TextFormField(
                                    initialValue:
                                        'XXXXXX ${scale.getMaxScaleOnAxis()}')))))
              ]))
        ]));
  }
}

```

**Comment by darshankawar** [comment]
Thanks for the update. I ran it on web and observed that upon first two clicks in the textfield, the offset values are different:

```
A Dart VM Service on Chrome is available at: http://127.0.0.1:50892/ttEsqW0bftY=
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:50892/ttEsqW0bftY=
Offset(187.3, 29.5)
Offset(93.6, 20.7)
```

Then subsequent clicks shows correct offsets as below:

```
Offset(44.2, 19.7)
Offset(44.2, 19.7)
Offset(44.1, 18.9)
Offset(44.1, 18.9)
```

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.7.12, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.7.12 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (3 days ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.11.0-1.0.pre.44, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.11.0-1.0.pre.44 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ebf3ea1755 (2 hours ago), 2023-05-03 23:25:08 -0400
    • Engine revision da45a21708
    • Dart version 3.1.0 (build 3.1.0-72.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.
      
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by jezell** [comment]
@darshankawar yes, that is because from then on you are clicking on the text host unless it loses focus. All of those values after the first are incorrect coordinates. Only the first one is the true coordinate.

**Comment by yjbanov** [comment]
Tentatively making it a P3, as it may actually be a regression either due to custom element embedding or the top-level DOM restructure to fix form autofill.

**Comment by htoor3** [comment]
Investigated this issue a bit and found a couple of things:
* It's a regression do to the DOM structure change. 
* Seems to be related to how we're computing the offsets on the engine `computeEventOffsetToTarget`
* In the test app, the correct offset and the calculated offset appear to be off by a factor of the `scale` transformation applied 
* Since text nodes have moved outside of the shadowDOM, we compute their offsets using the platform view offset logic `_computeOffsetRelativeToActualTarget`.  This function doesn't support computing offsets for transformed elements (see https://github.com/flutter/flutter/issues/117091), which I suspect is the issue. 
* The reason this seems to be fixed with setting `pointer-events: none` on the text editing host node is because in that case, we don't enter the platform view offset conditional because the `target` and `actualTarget` are the same `flutterViewElement`.  So it ends up using `ui.Offset(event.offsetX, event.offsetY)` which works with transforms of the parent element. 

I'm curious as to what side effects might occur by setting `pointer-events: none` on the parent element and letting the compute logic fall through to the case that supports transforms.  Main thing I would suspect is that the pointer clicks would register inaccurately, but basic testing shows that the offsets still calculate correctly. Some quick testing shows that keyboard still pops up as well.  

I can do some more extensive testing to see if that's a viable solution.  The other option would be support transforms for the platform view offset calculations and leave the text editing host as is.

**Comment by ditman** [comment]
A fix is coming to Flutter `master` soon: https://github.com/flutter/flutter/pull/127233 (we're also attempting to cherry-pick so it becomes available to everybody on `stable` (3.10)
